### PR TITLE
[REVIEW] Add numba modulo kernel for hash_encode

### LINKED ISSUE #935: [FEA] Replace numpy mod op in `Series.hash_encode`
The current implementation of `hash_encode` is bottlenecked at the numpy call https://github.com/rapidsai/cudf/blob/branch-0.6/python/cudf/dataframe/series.py#L1078. This computation should be moved to the gpu.

### PR DESCRIPTION
**Pull request by cmgreen210**
The current implementation of `Series.hash_encode`
uses `np.mod` on a `NumericalColumn`. This is extremely
slow so we add a numba kernel to speed this up.

Closes #935.

### CODE REVIEW

File: python/cudf/dataframe/series.py [hunk 1]
```diff
@@ -1075,7 +1075,10 @@ def hash_encode(self, stop, use_name=False):
         initial_hash = np.asarray(hash(self.name)) if use_name else None
         hashed_values = numerical.column_hash_values(
             self._column, initial_hash_values=initial_hash)
-        hashed_values = np.mod(hashed_values, stop)
+
+        # TODO: Binary op when https://github.com/rapidsai/cudf/pull/892 merged
+        cudautils.modulo(hashed_values.data.to_gpu_array(), stop)
```
Comment (kkraus14) [inline python/cudf/dataframe/series.py]: `hashed_values.data.to_gpu_array()` will return by copy, so you're writing into an ephemeral variable here

File: python/cudf/utils/cudautils.py [hunk 2]
```diff
@@ -727,3 +727,15 @@ def row_matrix(cols, nrow, ncol, dtype):
...
+
+
+@cuda.jit
+def gpu_modulo(arr, d):
+    i = cuda.grid(1)
+    if i < arr.size:
+        arr[i] %= d
+
+
+def modulo(arr, d):
```
Comment (kkraus14) [inline python/cudf/utils/cudautils.py]: I wouldn't expect this and the corresponding kernel to always run inplace, I think you should have an input array and an output array, along with the scalar.

File: python/cudf/dataframe/series.py (hunk 1 above)
Comment (cmgreen210) [reply python/cudf/dataframe/series.py]: If this were the case then shouldn't the test case fail: https://github.com/rapidsai/cudf/blob/1c0cd96b077c2703fe90fc99f5b8b916a672b04e/python/cudf/tests/test_dataframe.py#L1543

Comment (kkraus14) [reply python/cudf/dataframe/series.py]: If it isn't then we likely have a bug. Additionally, calling `to_gpu_array()` will skip null values by default so this won't work if you have a column of all nulls.

**Comment by jrhemstad** [comment]
This will just need to be replaced when the Jitify Binary ops are added, yes?

**Comment by kkraus14** [comment]
> This will just need to be replaced when the Jitify Binary ops are added, yes?

Yes, this isn't a full binary op and is more used as an internal function right now, but it could easily be replaced with a binary op once it exists 😄

**Pull request merged**